docs/DdnRelocatableShaderElf.md

```diff
@@ -0,0 +1,205 @@
+# Unlinked Shaders And Relocatable Shader Elf
+
+# INTRODUCTION
+
+Unlinked and relocatable shaders were designed with the intent of allowing
+offline compilation of shaders that will be linked together during pipeline
+creation. Compiling shaders offline means that the they can be compiled without
+running the vulkan application. The shaders will also be compiled individually.
+This is advantagous when the same shader is used in multiple pipelines because
+it will only be compiled once. It can also reduce the compilation time while the
+application is running.
```

Comment:
Do we want to qualify that this benefit can also be achieved with an in-memory driver cache, that the link time is not free but can be mitigated by an in-memory full-pipeline cache? Maybe the real advantage is to have the same benefits as an intermediate in-memory shader cache without the memory cost.

Reply:
The "real benefit" depends on how it is used, and it can be used in different ways.  I've updated this paragraph with a few different ways in which can help.